restore.go

```diff
@@ -137,7 +138,10 @@ func restoreContainer(context *cli.Context, spec *specs.Spec, config *configs.Co
 		if err != nil {
 			return -1, err
 		}
+		// TODO: restore should not create this fifo in the first place.
+		cleanupExecFifo(context)
```

Comment:
Yeah, I agree this a bit ugly. I think the problem boils down to the fact that `libcontainer.Factory.Create` is being used as both a "real" create (for `runc create`) but also as a "fake" create (in `runc restore`) in order to effectively make it look like `.Load` worked.

Maybe there's a nicer way of handling this? Should we move the stateful stuff out of `Create` into something separate? I'm not sure.